src/main/java/com/google/cloud/spanner/pgadapter/statements/local/SelectCurrentDatabaseStatement.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.cloud.spanner.pgadapter.statements.local;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.ResultSets;
+import com.google.cloud.spanner.Struct;
+import com.google.cloud.spanner.Type;
+import com.google.cloud.spanner.Type.StructField;
+import com.google.cloud.spanner.connection.StatementResult;
+import com.google.cloud.spanner.pgadapter.statements.BackendConnection;
+import com.google.cloud.spanner.pgadapter.statements.BackendConnection.QueryResult;
+import com.google.common.collect.ImmutableList;
+
+@InternalApi
+public class SelectCurrentDatabaseStatement implements LocalStatement {
+  public static final SelectCurrentDatabaseStatement INSTANCE =
+      new SelectCurrentDatabaseStatement();
+
+  private SelectCurrentDatabaseStatement() {}
+
+  @Override
+  public String[] getSql() {
+    return new String[] {
+      "select current_database()",
+      "SELECT current_database()",
+      "Select current_database()",
+      "select current_database",
+      "SELECT current_database",
+      "Select current_database",
+      "select * from current_database()",
+      "SELECT * FROM current_database()",
```

Comment:
Why do we need to write `"select current_database"` and `"select current_database"` as two different entries here?

Reply:
These statements are intended to support auto-generated queries from frameworks (e.g. ORMs). These could use different casings, although something like `sElEcT ...` is very unlikely to be used by any framework. So to keep it as efficient as possible, this has been created using a simple `Map` where the same statement is mapped to a number of different keys (one key for each casing that is reasonable). This has a much quicker lookup time than having to do a case-insensitive comparison of SQL strings. So you're right that this will not work for all possible ways to write the statement (it will for example also fail if the statement uses tabs instead of normal whitespaces to separate keywords), but that is an intended trade-off for efficiency in this case.

I've added a comment to the `LocalStatement` interface to clarify this.